WordPress/Sniffs/Theme/NoDeregisterCoreScriptSniff.php

```diff
@@ -0,0 +1,230 @@
+			'tiny_mce'                 => true,
+			'autosave'                 => true,
+			'wp-ajax-response'         => true,
+			'wp-lists'                 => true,
+			'common'                   => true,
+			'editorremov'              => true,
+			'editor-functions'         => true,
+			'ajaxcat'                  => true,
+			'admin-categories'         => true,
+			'admin-tags'               => true,
+			'admin-custom-fields'      => true,
+			'password-strength-meter'  => true,
+			'admin-comments'           => true,
+			'admin-users'              => true,
+			'admin-forms'              => true,
+			'xfn'                      => true,
+			'upload'                   => true,
+			'postbox'                  => true,
+			'slug'                     => true,
+			'post'                     => true,
+			'page'                     => true,
+			'link'                     => true,
+			'comment'                  => true,
+			'comment-reply'            => true,
+			'admin-gallery'            => true,
+			'media-upload'             => true,
+			'admin-widgets'            => true,
+			'word-count'               => true,
+			'theme-preview'            => true,
+			'json2'                    => true,
+			'plupload'                 => true,
+			'plupload-all'             => true,
+			'plupload-html4'           => true,
+			'plupload-html5'           => true,
+			'plupload-flash'           => true,
+			'plupload-silverlight'     => true,
+			'underscore'               => true,
+			'backbone'                 => true,
+			'imagesloaded'             => true,
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 0.xx.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		if ( ! isset( $parameters[1] ) ) {
+			return;
+		}
+
+		$matched_parameter = $this->strip_quotes( $parameters[1]['raw'] );
+
+		if ( isset( $this->target_functions[ $matched_content ][ $matched_parameter ] ) ) {
+			$this->throw_prohibited_error( $stackPtr, array( $matched_parameter ) );
+			return;
+		}
+
+		// More extensive check to prevent people circumventing the sniff.
+		$text                 = '';
+		$found_variable_token = false;
+
+		for ( $i = $parameters[1]['start']; $i <= $parameters[1]['end']; $i++ ) {
+			if ( isset( Tokens::$stringTokens[ $this->tokens[ $i ]['code'] ] ) === false ) {
+				if ( T_VARIABLE === $this->tokens[ $i ]['code']
+					|| T_STRING === $this->tokens[ $i ]['code']
+				) {
+					$found_variable_token = true;
+				}
+
+				continue;
+			}
+
+			$text .= $this->strip_quotes( $this->tokens[ $i ]['content'] );
+		}
```

Comment:
We were discussing in https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1364 about a utility function to return a string of all of the text string tokens.

I was wondering if it would be better to split the `$found_variable_token` from the `$text` code. This change would be my suggestion. https://gist.github.com/grappler/10dc57b4b3ab71a35e2fe925ef4da1ec The Unit Tests pass with this change.

Reply:
When I wrote that code, I chose not to do so for efficiency reasons. The `findNext()` function walks the token array as well, so with your example code, there would be two `foreach()`-es walking the same tokens, while the current code does the same while only walking the tokens once.